api/v1alpha1/apidefinition_webhook_test.go

```diff
@@ -239,3 +242,300 @@ func hasError(errs field.ErrorList, needle string) bool {
 
 	return false
 }
+
+func TestApiDefinition_Validate_Auth(t *testing.T) {
+	is := is.New(t)
+
+	tests := map[string]struct {
+		ApiDefinition ApiDefinition
+		ReturnErr     bool
```

Comment:
Do we need `ReturnErr` field to be present in the test case struct? Can't we just compare `ErrCause` with the `err` returned by `ApiDefinition.validate()`?